File: src/my_cpp_pkg/src/number_publisher.cpp
#include "rclcpp/rclcpp.hpp" //importing the ros2 lib if it shows a error open vs from the terminal
#include "example_interfaces/msg/int64.hpp"
using namespace std::chrono_literals;

class Number_Publisher : public rclcpp::Node //create a class and inharet from the rclcpp
{
    public: //puplic
        Number_Publisher() : Node("Number_Publisher") 
        {
            publisher_=this->create_publisher<example_interfaces::msg::Int64>("CPP_number",10);
            timer_ = this->create_wall_timer(1.0s,std::bind(&Number_Publisher::publisher,this));
            RCLCPP_INFO(this->get_logger(),"Number Publisher started...");
            

        }   
    private:
    //define the v of the class and the private methods
    rclcpp::Publisher<example_interfaces::msg::Int64>::SharedPtr publisher_;
    rclcpp::TimerBase::SharedPtr timer_;

    void publisher ()
    {
        auto msg_ = example_interfaces::msg::Int64();
        msg_.data = 2;
        this->publisher_->publish(msg_);
    }
};

int main(int argc,char **argv)
{
    rclcpp::init(argc,argv);
    auto node = std::make_shared<Number_Publisher>(); //create the shared pointer of the my node
    rclcpp::spin(node);
    rclcpp::shutdown();
    return 0;

}

File: src/my_py_pkg/my_py_pkg/number_publisher.py
#!/usr/bin/env python3
import rclpy
from rclpy.node import Node
from example_interfaces.msg import Int64

class Number_Publisher(Node): #create the class and inharet from node
    def __init__(self): 
        super().__init__("Number_Publisher") #call the init from the Node class
        self.publisher_=self.create_publisher(Int64,"number",10)
        self.timer_=self.create_timer(0.5,self.Publish_number)
        self.msg = Int64() #don't forget to treat this as class not varible
        self.msg.data= 2
        self.get_logger().info("Number_Publisher node is running.....")

    def Publish_number(self):
        self.publisher_.publish(self.msg)


def main(args=None):
    rclpy.init(args=args)   
    node=Number_Publisher()   
    rclpy.spin(node)      
    rclpy.shutdown()        



#if the name main called in the terminal call the main function
if __name__ == "__main__":
    main()